File: src/main/java/dev/mazurkiewicz/quizer/question/domain/port/QuestionRepository.java
package dev.mazurkiewicz.quizer.question.domain.port;

import dev.mazurkiewicz.quizer.exam.domain.port.QuestionNumber;
import dev.mazurkiewicz.quizer.question.domain.model.Question;
import dev.mazurkiewicz.quizer.question.domain.model.QuestionId;
import dev.mazurkiewicz.quizer.question.domain.model.QuestionInfo;

import java.util.List;

public interface QuestionRepository {

    Question getQuestionById(QuestionId id);

    List<Question> getRandomQuestions(QuestionNumber questionNumber);

    Question getRandomQuestion();

    QuestionInfo getQuestionInfo();
}

File: src/main/kotlin/io/mazurkiewicz/quizer/question/domain/port/QuestionRepository.kt
package io.mazurkiewicz.quizer.question.domain.port

import io.mazurkiewicz.quizer.question.domain.model.Question
import io.mazurkiewicz.quizer.question.domain.model.QuestionId

interface QuestionRepository {

    fun findQuestionById(id: QuestionId): Question

    fun saveQuestion(question: Question)

}